### PR TITLE
Version bump of net-im/turses to 0.2.23

### PR DESCRIPTION
**Pull request by alessandro-molari**
See https://bugs.gentoo.org/show_bug.cgi?id=532312

### CODE REVIEW

File: net-im/turses/turses-0.2.23.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2014 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: /var/cvsroot/gentoo-x86/net-im/turses/turses-0.2.13-r1.ebuild,v 1.2 2014/08/05 18:34:08 mrueg Exp $
+
+EAPI=5
+
+PYTHON_COMPAT=( python{2_6,2_7} )
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: please drop version 2.6

File: net-im/turses/turses-0.2.23.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+SRC_URI="https://github.com/alejandrogomez/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="
+	dev-python/oauth2[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: this should be build time only -> move to DEPEND

File: net-im/turses/turses-0.2.23.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	dev-python/tweepy[${PYTHON_USEDEP}]
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: Needs to be

```
"tweepy >2.2,<3"
```

File: net-im/turses/turses-0.2.23.ebuild [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	dev-python/urwid[${PYTHON_USEDEP}]
+"
+DEPEND="${RDEPEND}
+	test? (
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/nose[${PYTHON_USEDEP}]
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: Upstream switched to pytest

File: net-im/turses/turses-0.2.23.ebuild [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	)
+"
+
+DOCS=( AUTHORS HISTORY.rst README.rst )
+
+python_test() {
+	nosetests || die "Tests fail with ${EPYTHON}"
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: see above.

**Comment by mgorny** [comment]
@alem0lars, are you still there?

**Comment by alessandro-molari** [comment]
@mgorny yeah, sry for the delay.. i'm going to fix these problems now...

**Comment by alessandro-molari** [comment]
I've fixed the problems above.

I've already tested the changes. With `test` feature the tests are correctly run. The other comments you pointed out are also fixed.

I've also added the documentation stuff (html with `doc` use and man page normally). I've checked and they are correctly built (using upstream method). However I wasn't able to install them (in /usr/share/doc, /usr/share/man) but I don't know if it's my machine setup problems or something else.

**Comment by idella** [comment]
Al,
nice progress not quite there yet.
1. The dev-python/tox[${PYTHON_USEDEP}] still needs to come out of the list of deps behind test?.
2. Re-peruse the ebuilds that have a doc build they have a matching setup in install phase
What you have done is attempted to do the install immediately after the build. Remember it's inside a  python_compile_all() phase which is a form of src_compile().  The install is done separately in python_install_all().
See this example:
pep8-1.5.6.ebuild
Note the build is coded in the 1 line.  The build of the man page is unconditional so remains as is, yielding
python_compile_all() {
        use doc && emake -C docs html
        emake -C docs man
}
(which is much too close to doing the work for you)
The install; The whole phase from pep8-1.5.6.ebuild can simply drop in. Simply add the line with the cmd 'doman' before or after the line distutils-r1_python_install_all.  The "|| die" is not required for the 'doman' since it's a helper function.
3. Also DOC= can be completely deleted. The eclass distutils-r1 has a default install function which covers these, 'just works'. Delete the var, re-run and observe the final installed state.

File: net-im/turses/turses-0.2.23.ebuild [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	<dev-python/tweepy-3[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	doc? (
+		dev-python/sphinx[${PYTHON_USEDEP}]
+	)
+	test? (
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/coverage[${PYTHON_USEDEP}]
+		dev-python/tox[${PYTHON_USEDEP}]
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild:34]: This is not needed.

File: net-im/turses/turses-0.2.23.ebuild [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	)
+"
+
+python_compile_all() {
+	if use doc; then
+		emake -C docs html
+		dodoc -r "docs/_build/html" || die
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: In EAPI=5 all portage helper functions like dodoc or doman die internally.

File: net-im/turses/turses-0.2.23.ebuild [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+	fi
+	
+	emake -C docs man
+	doman "docs/_build/man/turses.1" || die
```
Comment (jlec) [inline net-im/turses/turses-0.2.23.ebuild]: see above.

**Comment by alessandro-molari** [comment]
I've updated the ebuild.

Now the documentation and man pages are correctly generated.

**Comment by jlec** [comment]
+*turses-0.2.23 (05 Jun 2015)
+
-  05 Jun 2015; Justin Lecher jlec@gentoo.org +turses-0.2.23.ebuild,
-  metadata.xml:
-  Version Bump, bug #532312; thanks Adam Jones and Alessandro Molari for the
-  ebuild, https://github.com/gentoo/gentoo-portage-rsync-mirror/pull/132
  +

**Pull request closed without merging**